test/e2e/kubeproxy.go

```diff
@@ -225,13 +228,17 @@ func (config *KubeProxyTestConfig) dialFromContainer(protocol, containerIP, targ
 		targetPort,
 		tries)
 
-	By(fmt.Sprintf("Dialing from container. Running command:%s", cmd))
-	stdout := config.ssh(cmd)
-	var output map[string][]string
-	err := json.Unmarshal([]byte(stdout), &output)
-	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Could not unmarshal curl response: %s", stdout))
-	hostNamesMap := array2map(output["responses"])
-	Expect(len(hostNamesMap)).To(BeNumerically("==", expectedCount), fmt.Sprintf("Response was:%v", output))
+	if providerIs(providersWithSSH...) {
```

Comment:
I like the idea of a net=host pod. net=host pods are disabled though AFAIK for every other pod config source than file. This can be overwritten by a kubelet flag. So you have to configure your cluster for net=host support.

Thoughts?